1.MLP_Binary_Classifier/nbconverted/Hyperparameter_Optimization_binary_class.py

```diff
@@ -0,0 +1,248 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Hyperparameter tuning via Optuna for Binary MLP model
+
+# ### Being a binary model this notebook will be limited to predicting one class 1 or 0, yes or no.
+# ### Here I will be predicting if a cell received a treatment or not
+
+# In[1]:
+
+
+import sys
+from pathlib import Path
+
+import matplotlib.pyplot as plt
+import numpy as np
+import optuna
+import pandas as pd
+import plotly
+import seaborn as sns
+import torch
+import torch.nn as nn
+import torch.optim as optim
+from sklearn import preprocessing
+from sklearn.metrics import (
+    accuracy_score,
+    auc,
+    classification_report,
+    confusion_matrix,
+    f1_score,
+    precision_score,
+    recall_score,
+    roc_auc_score,
+    roc_curve,
+)
+from sklearn.model_selection import train_test_split
+
+sys.path.append("..")
+from MLP_utils.parameters import Parameters
+
+# from MLP_utils.utils import build_model_custom
+# from MLP_utils.utils import train_n_validate
+from MLP_utils.utils import (
+    Dataset,
+    data_split,
+    extract_best_trial_params,
+    objective,
+    optimized_model,
+    plot_metric_vs_epoch,
+    results_output,
+    test_optimized_model,
+    train_optimized_model,
+    un_nest,
+)
+
+from utils.utils import df_stats
+
+params = Parameters()
+
+
+# In[2]:
+
+
+# Import Data
+# set data file path under pathlib path for multi-system use
+file_path = Path(
+    "../../Extracted_Features_(CSV_files)/interstellar_wave3_sc_norm_cellprofiler.csv.gz"
+)
+df = pd.read_csv(
+    file_path,
+    low_memory=False,
+)
+
+
+# Combine treatment with dosage to be able to discern treatments with different doses as a different condition
+
+# In[3]:
+
+
+# Combine treatment and dose
+df["Metadata_treatment"] = df["Metadata_treatment"] + "_" + df["Metadata_dose"]
```

Comment:
you'll probably want to create a new column, instead of overwriting an existing one.

Consider using [`pandas.DataFrame.assign()`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.assign.html)